Tcl 9.0b3 compat

```
$ git rev-parse HEAD
a53587eacd92d5108b846620be684e025ca5827f
```

```
$ gcc --version
gcc (GCC) 14.1.1 20240720
```

```
$ CFLAGS="-Og -g3 -Wall -D_FORTIFY_SOURCE=3" LD_LIBRARY_PATH=/home/michael/opt/tcl9.0b3/lib ./configure --with-tcllib=/home/michael/opt/tcl9.0b3/lib/libtcl9.0.so --with-tclinc=/home/michael/opt/tcl9.0b3/include/tcl.h 

This is Eggdrop's GNU configure script.
It's going to run a bunch of tests to hopefully make your compile
work without much twiddling.

checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for wchar.h... yes
checking for minix/config.h... no
checking for sys/select.h... yes
checking for sys/socket.h... yes
checking for sys/param.h... yes
checking whether it is safe to define __EXTENSIONS__... yes
checking whether _XOPEN_SOURCE should be defined... no
checking for library containing strerror... none required
checking for icc... no
checking whether the compiler understands -pipe... yes
checking whether the compiler understands -Wall... yes
checking for a BSD-compatible install... /usr/bin/install -c
checking for make... make
checking whether make sets $(MAKE)... yes
checking for ranlib... ranlib
checking whether ln -s works... yes
checking whether head -n 1 works... yes
checking for strip... strip
checking for gawk... gawk
checking for basename... basename
checking for uname... uname
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking for a sed that does not truncate output... /usr/bin/sed
checking how to run the C preprocessor... gcc -E
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for library containing gethostbyname... none required
checking for library containing socket... none required
checking whether gcc is Clang... no
checking whether pthreads work with "-pthread" and "-lpthread"... yes
checking for joinable pthread attribute... PTHREAD_CREATE_JOINABLE
checking whether more special flags are required for pthreads... no
checking for PTHREAD_PRIO_INHERIT... yes
checking for library containing dlopen... none required
checking for tan in -lm... yes
checking target system type... x86_64-pc-linux-gnu
checking system type... Linux
checking system release... 6.10.2-arch1-1
checking module loading capabilities... 
checking for dl.h... no
checking for dlfcn.h... yes
checking for loader.h... no
checking for rld.h... no
checking for mach-o/dyld.h... no
checking for mach-o/rld.h... no
checking for dlopen... yes
checking for load... no
checking for NSLinkModule... no
checking for shl_load... no
checking for rld_load... no
checking for dirent.h that defines DIR... yes
checking for library containing opendir... none required
checking for arpa/inet.h... yes
checking for fcntl.h... yes
checking for limits.h... yes
checking for locale.h... yes
checking for netdb.h... yes
checking for netinet/in.h... yes
checking for stdio.h... (cached) yes
checking for stdarg.h... yes
checking for stddef.h... yes
checking for sys/file.h... yes
checking for sys/param.h... (cached) yes
checking for sys/select.h... (cached) yes
checking for sys/socket.h... (cached) yes
checking for sys/time.h... yes
checking for time.h... yes
checking for unistd.h... (cached) yes
checking for wchar.h... (cached) yes
checking for an ANSI C-conforming const... yes
checking whether byte ordering is bigendian... no
checking for inline... inline
checking for flexible array members... yes
checking for intptr_t... yes
checking for pid_t... yes
checking for size_t... yes
checking whether struct tm is in sys/time.h or time.h... time.h
checking for gcc options needed to detect all undeclared functions... none needed
checking for struct tm.tm_zone... yes
checking for working volatile... yes
checking for socklen_t... yes
checking for stdint types... stdint.h (shortcircuit)
make use of stdint.h in eggint.h (assuming C99 compatible system)
checking for clock_gettime... yes
checking for dprintf... yes
checking for explicit_bzero... yes
checking for memset_explicit... no
checking for explicit_memset... no
checking for getrandom... yes
checking for inet_aton... yes
checking for memset_s... no
checking for snprintf... yes
checking for strlcpy... yes
checking for vsnprintf... yes
checking types of arguments for select... int,fd_set *,struct timeval *
checking for b64_ntop... no
checking for b64_ntop with -lresolv... yes
checking for getpagesize... yes
checking for working mmap... yes
checking for Tcl linker... ${CC} ${CFLAGS} ${LDFLAGS} -shared
checking for Tcl version... 9.0b3
checking for Tcl library flags... -L/home/michael/opt/tcl9.0b3/lib -ltcl9.0  -lz -lpthread -lm -ltommath
checking for Tcl header flags... -I/home/michael/opt/tcl9.0b3/include
checking whether C compiler accepts -Og... yes
checking whether C compiler accepts -fsanitize=address... yes
checking for system IPv6 support... yes
checking for struct in6_addr... yes
checking for the in6addr_any constant... yes
checking for the in6addr_loopback constant... yes
checking for struct sockaddr_in6... yes
checking whether to enable TLS support... yes
checking for openssl/ssl.h... yes
checking for openssl/x509v3.h... yes
checking for X509_digest in -lcrypto... yes
checking for SSL_accept in -lssl... yes
checking for EVP_sha1... yes
checking for a2i_IPADDRESS... yes
checking for EVP_md5... yes
checking for OPENSSL_buf2hexstr... yes
checking for OPENSSL_hexstr2buf... yes
checking for OpenSSL... yes
checking for ASN1_STRING_get0_data... yes
checking for EVP_PKEY_get1_EC_KEY... yes
checking for openssl/md5.h... yes
checking for MD5_Init... yes
checking for MD5_Update... yes
checking for MD5_Final... yes
configure: creating ./config.status
config.status: creating Makefile
config.status: creating doc/Makefile
config.status: creating scripts/Makefile
config.status: creating src/Makefile
config.status: creating src/compat/Makefile
config.status: creating src/md5/Makefile
config.status: creating src/mod/Makefile
config.status: creating config.h
config.status: executing eggint.h commands
config.status: creating eggint.h : __EGGINT_H
config.status: executing replace-if-changed commands
creating lush.h
config.status: executing catch-make-rebuild commands

Operating System: Linux 6.10.2-arch1-1
IPv6 Support: yes
Tcl version: 9.0b3
SSL/TLS Support: yes (OpenSSL 3.3.1 4 Jun 2024)
Threaded DNS core: yes

If you experience any problems compiling Eggdrop, please read the
compile guide, found in doc/COMPILE-GUIDE.

Type 'make config' to configure the modules, or type 'make iconfig'
to interactively choose which modules to compile.
```

```
$ LD_LIBRARY_PATH=/home/michael/opt/tcl9.0b3/lib make config
Detecting modules....................... done.
Calculating dependencies...................... done.
Building ./src/mod/Makefile... done.
make[1]: Entering directory '/home/michael/projects/eggdrop/src/mod'

Configuring module 'compress'.
configure: creating cache ../../../config.cache
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking for gzopen in -lz... yes
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for zlib.h... yes
configure: updating cache ../../../config.cache
configure: creating ./config.status
config.status: creating Makefile

Configuring module 'dns'.
configure: loading cache ../../../config.cache
checking for gcc... (cached) gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... (cached) o
checking whether the compiler supports GNU C... (cached) yes
checking whether gcc accepts -g... (cached) yes
checking for gcc option to enable C11 features... (cached) none needed
checking for sys/types.h... (cached) yes
checking for netinet/in.h... yes
checking for arpa/nameser.h... yes
checking for netdb.h... yes
checking for resolv.h... yes
checking for a working resolver library... yes ()
configure: updating cache ../../../config.cache
configure: creating ./config.status
config.status: creating Makefile

Configuring module 'python'.
configure: loading cache ../../../config.cache
checking for grep that handles long lines and -e... /usr/bin/grep
checking for fgrep... /usr/bin/grep -F
checking for python3-config... /usr/bin/python3-config
checking whether python-config supports --embed... yes
checking for python C flags... -I/usr/include/python3.12 -I/usr/include/python3.12  -fno-strict-overflow -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection         -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -ffat-lto-objects -DNDEBUG -g -O3 -Wall
checking for python LD flags...  -L/usr/lib -lpython3.12 -ldl  -lm 
checking for gawk... gawk
checking python version is >= 3.8.0... yes (3.12)
configure: updating cache ../../../config.cache
configure: creating ./config.status
config.status: creating Makefile

make[1]: Leaving directory '/home/michael/projects/eggdrop/src/mod'
Building ./src/mod/Makefile... done.

You can now compile the bot, using "make".
```

```
$ LD_LIBRARY_PATH=/home/michael/opt/tcl9.0b3/lib make eggdrop

Making core eggdrop files...

make[1]: Entering directory '/home/michael/projects/eggdrop/src'
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I.. -I..  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -c bg.c
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I.. -I..  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -c botcmd.c
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I.. -I..  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -c botmsg.c
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I.. -I..  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -c botnet.c
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I.. -I..  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -c chanprog.c
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I.. -I..  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -c cmds.c
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I.. -I..  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -c dcc.c
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I.. -I..  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -c dccutil.c
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I.. -I..  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -c dns.c
dns.c: In function ‘thread_dns_ipbyhost’:
dns.c:571:99: warning: ‘%s’ directive output may be truncated writing up to 2047 bytes into a region of size 147 [-Wformat-truncation=]
  571 |     snprintf(dtn->strerror, sizeof dtn->strerror, "dns: thread_dns_ipbyhost(): getaddrinfo(): %s: %s", gai_strerror(error), ebuf);
      |                                                                                                   ^~                        ~~~~
In file included from /usr/include/stdio.h:970,
                 from /home/michael/opt/tcl9.0b3/include/tcl.h:106,
                 from ../lush.h:4,
                 from main.h:42,
                 from dns.c:27:
In function ‘snprintf’,
    inlined from ‘thread_dns_ipbyhost’ at dns.c:571:5:
/usr/include/bits/stdio2.h:68:10: note: ‘__builtin___snprintf_chk’ output 46 or more bytes (assuming 2093) into a destination of size 192
   68 |   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   69 |                                    __glibc_objsize (__s), __fmt,
      |                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   70 |                                    __va_arg_pack ());
      |                                    ~~~~~~~~~~~~~~~~~
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I.. -I..  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -c flags.c
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I.. -I..  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -c language.c
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I.. -I..  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -c match.c
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I.. -I..  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  \
'-DCCFLAGS="gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I.. -I..  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include "' \
'-DLDFLAGS="gcc -lpthread"' \
'-DSTRIPFLAGS="touch"' -c ./main.c
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I.. -I..  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -c mem.c
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I.. -I..  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -c misc.c
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I.. -I..  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -c misc_file.c
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I.. -I..  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -c modules.c
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I.. -I..  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -c net.c
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I.. -I..  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -c rfc1459.c
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I.. -I..  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -c tcl.c
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I.. -I..  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -c tcldcc.c
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I.. -I..  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -c tclhash.c
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I.. -I..  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -c tclmisc.c
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I.. -I..  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -c tcluser.c
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I.. -I..  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -c tls.c
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I.. -I..  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -c userent.c
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I.. -I..  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -c userrec.c
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I.. -I..  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -c users.c
make[2]: Entering directory '/home/michael/projects/eggdrop/src/md5'
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I. -I../.. -I../.. -I../../src  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -c md5c.c
make[2]: Leaving directory '/home/michael/projects/eggdrop/src/md5'
make[2]: Entering directory '/home/michael/projects/eggdrop/src/compat'
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I../.. -I../.. -I../../src  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -c base64.c
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I../.. -I../.. -I../../src  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -c explicit_bzero.c
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I../.. -I../.. -I../../src  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -c in6.c
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I../.. -I../.. -I../../src  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -c inet_aton.c
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I../.. -I../.. -I../../src  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -c snprintf.c
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I../.. -I../.. -I../../src  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -c strlcpy.c
make[2]: Leaving directory '/home/michael/projects/eggdrop/src/compat'
make linkstart && make link && make linkfinish
make[2]: Entering directory '/home/michael/projects/eggdrop/src'

This may take a while. Go get some runts.

---------- Yeah! That's the compiling, now the linking! ----------

Linking eggdrop (standard build).

make[2]: Leaving directory '/home/michael/projects/eggdrop/src'
make[2]: Entering directory '/home/michael/projects/eggdrop/src'
gcc -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I.. -I..  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include -lpthread -o ../eggdrop bg.o botcmd.o botmsg.o botnet.o chanprog.o cmds.o dcc.o dccutil.o dns.o flags.o language.o match.o main.o mem.o misc.o misc_file.o modules.o net.o rfc1459.o tcl.o tcldcc.o tclhash.o tclmisc.o tcluser.o tls.o userent.o userrec.o users.o  -L/home/michael/opt/tcl9.0b3/lib -ltcl9.0  -lz -lpthread -lm -ltommath -lssl -lcrypto  -lresolv md5/md5c.o compat/*.o `cat mod/mod.xlibs`
/usr/bin/ld: /usr/lib/libasan.so.8: warning: the use of `tmpnam_r' is dangerous, better use `mkstemp'
/usr/bin/ld: /usr/lib/libasan.so.8: warning: the use of `tempnam' is dangerous, better use `mkstemp'
/usr/bin/ld: /usr/lib/libasan.so.8: warning: the use of `tmpnam' is dangerous, better use `mkstemp'
make[2]: Leaving directory '/home/michael/projects/eggdrop/src'
make[2]: Entering directory '/home/michael/projects/eggdrop/src'

Successful compile: eggdrop

make[2]: Leaving directory '/home/michael/projects/eggdrop/src'
make[1]: Leaving directory '/home/michael/projects/eggdrop/src'

Making modules...

make[1]: Entering directory '/home/michael/projects/eggdrop/src/mod'
make[2]: Entering directory '/home/michael/projects/eggdrop/src/mod/assoc.mod'
gcc -fPIC -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I. -I../../.. -I../../.. -I../../../src/mod  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -DMAKING_MODS -c .././assoc.mod/assoc.c && mv -f assoc.o ../
gcc   -shared -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I. -I../../.. -I../../.. -I../../../src/mod  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include -o ../../../assoc.so ../assoc.o -L/home/michael/opt/tcl9.0b3/lib -ltcl9.0  -lz -lpthread -lm -ltommath -lssl -lcrypto  -lresolv  && touch ../../../assoc.so
make[2]: Leaving directory '/home/michael/projects/eggdrop/src/mod/assoc.mod'
make[2]: Entering directory '/home/michael/projects/eggdrop/src/mod/blowfish.mod'
gcc -fPIC -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I. -I../../.. -I../../.. -I../../../src/mod  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -DMAKING_MODS -c .././blowfish.mod/blowfish.c && mv -f blowfish.o ../
gcc   -shared -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I. -I../../.. -I../../.. -I../../../src/mod  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include -o ../../../blowfish.so ../blowfish.o -L/home/michael/opt/tcl9.0b3/lib -ltcl9.0  -lz -lpthread -lm -ltommath -lssl -lcrypto  -lresolv  && touch ../../../blowfish.so
make[2]: Leaving directory '/home/michael/projects/eggdrop/src/mod/blowfish.mod'
make[2]: Entering directory '/home/michael/projects/eggdrop/src/mod/channels.mod'
gcc -fPIC -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I. -I../../.. -I../../.. -I../../../src/mod  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -DMAKING_MODS -c .././channels.mod/channels.c && mv -f channels.o ../
gcc   -shared -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I. -I../../.. -I../../.. -I../../../src/mod  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include -o ../../../channels.so ../channels.o -L/home/michael/opt/tcl9.0b3/lib -ltcl9.0  -lz -lpthread -lm -ltommath -lssl -lcrypto  -lresolv  && touch ../../../channels.so
make[2]: Leaving directory '/home/michael/projects/eggdrop/src/mod/channels.mod'
make[2]: Entering directory '/home/michael/projects/eggdrop/src/mod/compress.mod'
gcc -fPIC -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I. -I../../.. -I../../.. -I../../../src/mod  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -DMAKING_MODS -c .././compress.mod/compress.c && mv -f compress.o ../
gcc   -shared -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I. -I../../.. -I../../.. -I../../../src/mod  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include -o ../../../compress.so ../compress.o -lz -L/home/michael/opt/tcl9.0b3/lib -ltcl9.0  -lz -lpthread -lm -ltommath -lssl -lcrypto  -lresolv  && touch ../../../compress.so
make[2]: Leaving directory '/home/michael/projects/eggdrop/src/mod/compress.mod'
make[2]: Entering directory '/home/michael/projects/eggdrop/src/mod/console.mod'
gcc -fPIC -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I. -I../../.. -I../../.. -I../../../src/mod  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -DMAKING_MODS -c .././console.mod/console.c && mv -f console.o ../
gcc   -shared -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I. -I../../.. -I../../.. -I../../../src/mod  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include -o ../../../console.so ../console.o -L/home/michael/opt/tcl9.0b3/lib -ltcl9.0  -lz -lpthread -lm -ltommath -lssl -lcrypto  -lresolv  && touch ../../../console.so
make[2]: Leaving directory '/home/michael/projects/eggdrop/src/mod/console.mod'
make[2]: Entering directory '/home/michael/projects/eggdrop/src/mod/ctcp.mod'
gcc -fPIC -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I. -I../../.. -I../../.. -I../../../src/mod  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -DMAKING_MODS -c .././ctcp.mod/ctcp.c && mv -f ctcp.o ../
gcc   -shared -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I. -I../../.. -I../../.. -I../../../src/mod  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include -o ../../../ctcp.so ../ctcp.o -L/home/michael/opt/tcl9.0b3/lib -ltcl9.0  -lz -lpthread -lm -ltommath -lssl -lcrypto  -lresolv  && touch ../../../ctcp.so
make[2]: Leaving directory '/home/michael/projects/eggdrop/src/mod/ctcp.mod'
make[2]: Entering directory '/home/michael/projects/eggdrop/src/mod/dns.mod'
gcc -fPIC -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I. -I../../.. -I../../.. -I../../../src/mod  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include   -DMAKING_MODS -c .././dns.mod/dns.c && mv -f dns.o ../
gcc   -shared -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I. -I../../.. -I../../.. -I../../../src/mod  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include -o ../../../dns.so ../dns.o  -L/home/michael/opt/tcl9.0b3/lib -ltcl9.0  -lz -lpthread -lm -ltommath -lssl -lcrypto  -lresolv  && touch ../../../dns.so
make[2]: Leaving directory '/home/michael/projects/eggdrop/src/mod/dns.mod'
make[2]: Entering directory '/home/michael/projects/eggdrop/src/mod/filesys.mod'
gcc -fPIC -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I. -I../../.. -I../../.. -I../../../src/mod  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -DMAKING_MODS -c .././filesys.mod/filesys.c && mv -f filesys.o ../
gcc   -shared -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I. -I../../.. -I../../.. -I../../../src/mod  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include -o ../../../filesys.so ../filesys.o -L/home/michael/opt/tcl9.0b3/lib -ltcl9.0  -lz -lpthread -lm -ltommath -lssl -lcrypto  -lresolv  && touch ../../../filesys.so
make[2]: Leaving directory '/home/michael/projects/eggdrop/src/mod/filesys.mod'
make[2]: Entering directory '/home/michael/projects/eggdrop/src/mod/ident.mod'
gcc -fPIC -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I. -I../../.. -I../../.. -I../../../src/mod  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -DMAKING_MODS -c .././ident.mod/ident.c && mv -f ident.o ../
gcc   -shared -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I. -I../../.. -I../../.. -I../../../src/mod  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include -o ../../../ident.so ../ident.o -L/home/michael/opt/tcl9.0b3/lib -ltcl9.0  -lz -lpthread -lm -ltommath -lssl -lcrypto  -lresolv  && touch ../../../ident.so
make[2]: Leaving directory '/home/michael/projects/eggdrop/src/mod/ident.mod'
make[2]: Entering directory '/home/michael/projects/eggdrop/src/mod/irc.mod'
gcc -fPIC -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I. -I../../.. -I../../.. -I../../../src/mod  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -DMAKING_MODS -c .././irc.mod/irc.c && mv -f irc.o ../
gcc   -shared -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I. -I../../.. -I../../.. -I../../../src/mod  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include -o ../../../irc.so ../irc.o -L/home/michael/opt/tcl9.0b3/lib -ltcl9.0  -lz -lpthread -lm -ltommath -lssl -lcrypto  -lresolv  && touch ../../../irc.so
make[2]: Leaving directory '/home/michael/projects/eggdrop/src/mod/irc.mod'
make[2]: Entering directory '/home/michael/projects/eggdrop/src/mod/notes.mod'
gcc -fPIC -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I. -I../../.. -I../../.. -I../../../src/mod  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -DMAKING_MODS -c .././notes.mod/notes.c && mv -f notes.o ../
gcc   -shared -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I. -I../../.. -I../../.. -I../../../src/mod  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include -o ../../../notes.so ../notes.o -L/home/michael/opt/tcl9.0b3/lib -ltcl9.0  -lz -lpthread -lm -ltommath -lssl -lcrypto  -lresolv  && touch ../../../notes.so
make[2]: Leaving directory '/home/michael/projects/eggdrop/src/mod/notes.mod'
make[2]: Entering directory '/home/michael/projects/eggdrop/src/mod/pbkdf2.mod'
gcc -fPIC -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I. -I../../.. -I../../.. -I../../../src/mod  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -DMAKING_MODS -c .././pbkdf2.mod/pbkdf2.c && mv -f pbkdf2.o ../
gcc   -shared -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I. -I../../.. -I../../.. -I../../../src/mod  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include -o ../../../pbkdf2.so ../pbkdf2.o -L/home/michael/opt/tcl9.0b3/lib -ltcl9.0  -lz -lpthread -lm -ltommath -lssl -lcrypto  -lresolv  && touch ../../../pbkdf2.so
make[2]: Leaving directory '/home/michael/projects/eggdrop/src/mod/pbkdf2.mod'
make[2]: Entering directory '/home/michael/projects/eggdrop/src/mod/python.mod'
gcc -fPIC -I/usr/include/python3.12 -I/usr/include/python3.12  -fno-strict-overflow -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection         -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -ffat-lto-objects -DNDEBUG -g -O3 -Wall -Og -g3 -Wall -D_FORTIFY_SOURCE=3 -pipe -Wall -I. -I../../.. -I../../.. -I../../../src/mod  -DHAVE_CONFIG_H -I/home/michael/opt/tcl9.0b3/include  -DMAKING_MODS -c .././python.mod/python.c && mv -f python.o ../
In file included from .././python.mod/python.c:46:
.././python.mod/pycmds.c: In function ‘py_parse_tcl_list’:
.././python.mod/pycmds.c:170:44: error: passing argument 3 of ‘Tcl_ListObjLength’ from incompatible pointer type [-Wincompatible-pointer-types]
  170 |   if (Tcl_ListObjLength(tclinterp, strobj, &max) != TCL_OK) {
      |                                            ^~~~
      |                                            |
      |                                            int *
In file included from /home/michael/opt/tcl9.0b3/include/tcl.h:2435,
                 from ../../../lush.h:4,
                 from ../../../src/main.h:42,
                 from ../../../src/mod/module.h:28,
                 from .././python.mod/python.c:29:
/home/michael/opt/tcl9.0b3/include/tclDecls.h:1789:61: note: expected ‘Tcl_Size *’ {aka ‘long int *’} but argument is of type ‘int *’
 1789 |                                 Tcl_Obj *listPtr, Tcl_Size *lengthPtr);
      |                                                   ~~~~~~~~~~^~~~~~~~~
.././python.mod/pycmds.c:181:43: error: passing argument 2 of ‘Tcl_GetStringFromObj’ from incompatible pointer type [-Wincompatible-pointer-types]
  181 |     tclstr = Tcl_GetStringFromObj(tclobj, &tclstrlen);
      |                                           ^~~~~~~~~~
      |                                           |
      |                                           int *
/home/michael/opt/tcl9.0b3/include/tclDecls.h:1754:43: note: expected ‘Tcl_Size *’ {aka ‘long int *’} but argument is of type ‘int *’
 1754 |                                 Tcl_Size *lengthPtr);
      |                                 ~~~~~~~~~~^~~~~~~~~
.././python.mod/pycmds.c: In function ‘py_parse_tcl_dict’:
.././python.mod/pycmds.c:207:54: error: passing argument 2 of ‘Tcl_GetStringFromObj’ from incompatible pointer type [-Wincompatible-pointer-types]
  207 |     const char *valstr = Tcl_GetStringFromObj(value, &len);
      |                                                      ^~~~
      |                                                      |
      |                                                      int *
/home/michael/opt/tcl9.0b3/include/tclDecls.h:1754:43: note: expected ‘Tcl_Size *’ {aka ‘long int *’} but argument is of type ‘int *’
 1754 |                                 Tcl_Size *lengthPtr);
      |                                 ~~~~~~~~~~^~~~~~~~~
make[2]: *** [Makefile:19: ../python.o] Error 1
make[2]: Leaving directory '/home/michael/projects/eggdrop/src/mod/python.mod'
make[1]: *** [Makefile:60: python.mod_so] Error 2
make[1]: Leaving directory '/home/michael/projects/eggdrop/src/mod'
make: *** [Makefile:243: eggdrop] Error 2
```

--- a/src/mod/python.mod/pycmds.c
+++ b/src/mod/python.mod/pycmds.c
@@ -156,7 +156,7 @@ static int tcl_call_python(ClientData cd, Tcl_Interp *irp, int objc, Tcl_Obj *co
 }
 
 static PyObject *py_parse_tcl_list(PyObject *self, PyObject *args) {
-  int max;
+  Tcl_Size max;
   const char *str;
   Tcl_Obj *strobj;
   PyObject *result;
@@ -175,7 +175,7 @@ static PyObject *py_parse_tcl_list(PyObject *self, PyObject *args) {
   for (int i = 0; i < max; i++) {
     Tcl_Obj *tclobj;
     const char *tclstr;
-    int tclstrlen;
+    Tcl_Size tclstrlen;
 
     Tcl_ListObjIndex(tclinterp, strobj, i, &tclobj);
     tclstr = Tcl_GetStringFromObj(tclobj, &tclstrlen);
@@ -203,7 +203,7 @@ static PyObject *py_parse_tcl_dict(PyObject *self, PyObject *args) {
   }
   result = PyDict_New();
   while (!done) {
-    int len;
+    Tcl_Size len;
     const char *valstr = Tcl_GetStringFromObj(value, &len);
     PyObject *pyval = PyUnicode_DecodeUTF8(valstr, len, NULL);
     PyDict_SetItemString(result, Tcl_GetString(key), pyval);

